src/Infrastructure.Styles.Analyzer/RMSYN1102SeparateExitConditionsCodeFixProvider.cs

```diff
@@ -0,0 +1,218 @@
+        blueprint = new IfAndElseIfStatementSyntaxManipulator(ifStatement.Parent, ifStatement.Else);
+      else
+        blueprint = new IfAndElseIfStatementSyntaxManipulator(ifStatement, ifStatement.Else);
+
+      return blueprint;
+    }
+
+    private static SyntaxNode BuildRoot (
+      SyntaxNode oldRoot,
+      List<IfAndElseIfStatementSyntaxManipulator> syntaxNodes,
+      IfAndElseIfStatementSyntaxManipulator oldNode)
+    {
+      BuildProperStatementList(syntaxNodes);
+
+      //ReplaceNode() can only replace an ifNode by a list if the list does not include else statements.
+      //Else if statements can only be in the list if it's count is 1
+      if (syntaxNodes.Count == 1)
+        return oldRoot.ReplaceNode(oldNode.IfOrElseIfStatement, syntaxNodes.First().IfOrElseIfStatement);
+
+      return oldRoot.ReplaceNode(
+        oldNode.IfOrElseIfStatement, 
+        syntaxNodes.Select(s => s.IfOrElseIfStatement));
+    }
+
+    private static void BuildProperStatementList (List<IfAndElseIfStatementSyntaxManipulator> syntaxNodes)
+    {
+      var tempElseClause = syntaxNodes.Last();
+      syntaxNodes.RemoveAt(syntaxNodes.Count - 1);
+
+      while (syntaxNodes.Count > 0)
+      {
+        if (tempElseClause.IfOrElseIfStatement.IsKind(SyntaxKind.IfStatement))
+          break;
+
+        var tempParentElseClause = syntaxNodes.Last();
+
+        tempParentElseClause.WithElse((ElseClauseSyntax)tempElseClause.IfOrElseIfStatement);
+        tempElseClause = tempParentElseClause;
+        syntaxNodes.RemoveAt(syntaxNodes.Count - 1);
+      }
+
+      syntaxNodes.Add(tempElseClause);
+    }
+
+
+    private static void FillSyntaxNodeListWithStatements (
+      List<IfAndElseIfStatementSyntaxManipulator> syntaxNodes,
+      IfAndElseIfStatementSyntaxManipulator blueprint,
+      SyntaxNode expression)
+    {
+      if (SimpleIfStatementDetectorUtility.IsIllegalExpression(expression, out var illegalExpression))
+      {
+        var splitExpression = SplitExpressionUtility.SplitExpression(illegalExpression!);
+        FillSyntaxNodeListWithStatements(syntaxNodes, blueprint, splitExpression.Left);
+        FillSyntaxNodeListWithStatements(syntaxNodes, blueprint, splitExpression.Right);
+        return;
+      }
+
+      blueprint.WithCondition(expression);
+      syntaxNodes.Add(blueprint.Copy());
+    }
+
+    private static void ReintroduceParentElseClauses (List<IfAndElseIfStatementSyntaxManipulator> syntaxNodes)
+    {
+      foreach (var (syntaxNode, index) in syntaxNodes.Select((syntaxNode, i) => (syntaxNode, i)))
+      {
+        if (index == 0)
+          continue;
+
+        syntaxNode.ReintroduceParentElseClause();
+      }
+    }
+
+    private static SyntaxNode FindTopMostParentExpression (SyntaxNode node)
+    {
+      if (node.Parent.IsKind(SyntaxKind.IfStatement))
+        return node;
+      if (node.Parent.IsKind(SyntaxKind.IsPatternExpression))
+        return node;
+
+      if (node.Parent == null)
+        return node;
```

Comment:
I think there should be no blank line here. the three exit conditions are equivalent.